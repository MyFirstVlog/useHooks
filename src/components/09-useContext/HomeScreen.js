import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {

    const {user, setuser} = useContext(UserContext)

    console.log(user)

    return (
        <div>
            <h1>Home Screen</h1>

            <pre>
                {JSON.stringify(user,null,3)}
            </pre>

        </div>
        
    )
}
