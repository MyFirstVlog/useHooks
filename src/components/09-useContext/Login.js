import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const Login = () => {

    const {setuser} = useContext(UserContext)

    const User = {
        id:1234,
        name:'Alejandro Estrada',
        email:'alesmo30@gmail.com'
    }

    return (
        <div>
            <h1>Login Screen</h1>

            <button 
                className='btn btn-primary'
                onClick = {()=> setuser(User)}
            >
                Login
            </button>
        </div>
    )
}

