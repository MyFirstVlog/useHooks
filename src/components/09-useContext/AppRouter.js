import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Routes,
  Redirect,
  Link,
  useRouteMatch,
  useParams
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from "./HomeScreen";
import { Login } from "./Login";
import { NavBar } from "./NavBar";


export const AppRouter = () => {

    //implementaciones cambiaron en el switch
    return (
        <Router>
            <div>

                <NavBar/>

                <Routes>
                    <Route exact path='/' element={<HomeScreen/>} />
                    <Route exact path='/about' element={<AboutScreen/>} />
                    <Route exact path='/login' element={<Login/>} />
                    <Route path="*" element={<HomeScreen />} />
                </Routes>
            </div>
        </Router>
    )
}

