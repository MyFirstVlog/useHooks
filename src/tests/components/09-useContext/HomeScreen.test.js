import { mount, shallow } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en <HomeScreen/>', () => {

    const user = {
        name: 'Alejandro',
        email: 'alesmo30@gmail.com'
    }

    const wrapper = mount(<UserContext.Provider value={{ //no sirve shallow porque solo renderiza el componente padre, para esto usamos mount
            user
        }}>
        <HomeScreen />
    </UserContext.Provider>)
    test('debe de mostar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
})
