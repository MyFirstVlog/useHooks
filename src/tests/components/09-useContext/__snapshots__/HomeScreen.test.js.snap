// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> debe de mostar el componente correctamente  1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <pre>
    {
   "name": "Alejandro",
   "email": "alesmo30@gmail.com"
}
  </pre>
</div>
`;
